src/serac/numerics/assembled_sparse_matrix.cpp

```diff
@@ -0,0 +1,274 @@
+
+#include "mfem/linalg/dtensor.hpp"
+#include "assembled_sparse_matrix.hpp"
+
+template <auto offsetV, auto indicesV, auto gatherV>
+struct forbidden_restriction {
+  //  friend result_fes _get_fes(From & from) { return & from.*fesV; }
+  friend mfem::Array<int>& __get_offsets(mfem::ElementRestriction& from) { return from.*offsetV; }
+  friend mfem::Array<int>& __get_indices(mfem::ElementRestriction& from) { return from.*indicesV; }
+  friend mfem::Array<int>& __get_gatherMap(mfem::ElementRestriction& from) { return from.*gatherV; }
+};
+
+// const FiniteElementSpace * __get_fes(mfem::ElementRestriction &);
+mfem::Array<int>& __get_offsets(mfem::ElementRestriction&);
+mfem::Array<int>& __get_indices(mfem::ElementRestriction&);
+mfem::Array<int>& __get_gatherMap(mfem::ElementRestriction&);
```

Comment:
It's unlikely to actually be an issue, but names starting with double underscore are technically reserved for the language implementation / compiler, so I think these should be renamed.